ui/app/services/permissions.js

```diff
@@ -0,0 +1,123 @@
+import Service, { inject as service } from '@ember/service';
+import { task, waitForProperty } from 'ember-concurrency';
+
+const API_PATHS = {
+  secrets: { engine: 'cubbyhole/' },
+  access: {
+    methods: 'sys/auth',
+    entities: 'identity/entities',
+    groups: 'identity/groups',
+    leases: 'sys/leases/lookup',
+    namespaces: 'sys/namespaces',
+    'control-groups': 'sys/control-group/',
+  },
+  policies: {
+    acl: 'sys/policies/acl',
+    rgp: 'sys/policies/rgp',
+    egp: 'sys/policies/egp',
+  },
+  tools: {
+    wrap: 'sys/wrapping/wrap',
+    lookup: 'sys/wrapping/lookup',
+    unwrap: 'sys/wrapping/unwrap',
+    rewrap: 'sys/wrapping/rewrap',
+    random: 'sys/tools/random',
+    hash: 'sys/tools/hash',
+  },
+  status: {
+    replication: 'sys/replication',
+    license: 'sys/license',
+    seal: 'sys/seal',
+  },
+};
+
+const API_PATHS_TO_ROUTE_PARAMS = {
+  'sys/auth': ['vault.cluster.access.methods'],
+  'identity/entities': ['vault.cluster.access.identity', 'entities'],
+  'identity/groups': ['vault.cluster.access.identity', 'groups'],
+  'sys/leases/lookup': ['vault.cluster.access.leases'],
+  'sys/namespaces': ['vault.cluster.access.namespaces'],
+  'sys/control-group/': ['vault.cluster.access.control-groups'],
+};
+
+export default Service.extend({
+  exactPaths: null,
+  globPaths: null,
+  isRootToken: null,
+  store: service(),
+  auth: service(),
+
+  getPaths: task(function*() {
+    if (this.paths) {
+      return;
+    }
+    let resp = yield this.get('store')
+      .adapterFor('permissions')
+      .query();
+    this.setPaths(resp);
+    return;
+  }),
+
+  setPaths(resp) {
+    this.set('exactPaths', resp.data.exact_paths);
+    this.set('globPaths', resp.data.glob_paths);
+    this.set('isRootToken', resp.data.root);
+  },
+
+  clearPaths() {
```

Comment:
What do you think about calling this `reset` since it's also re-starting the checkAuthToken task?

Reply:
i like it!